rpcserver.go

```diff
@@ -3437,6 +3501,16 @@ func newRPCServer(listenAddrs []string, s *server) (*rpcServer, error) {
 		helpCacher:   newHelpCacher(),
 		quit:         make(chan int),
 	}
+	if cfg.RPCUser != "" && cfg.RPCPass != "" {
+		login := cfg.RPCUser + ":" + cfg.RPCPass
+		auth := "Basic " + base64.StdEncoding.EncodeToString([]byte(login))
+		rpc.authsha = fastsha256.Sum256([]byte(auth))
+	}
+	if cfg.RPCLimitUser != "" && cfg.RPCLimitPass != "" {
+		login := cfg.RPCLimitUser + ":" + cfg.RPCLimitPass
+		roauth := "Basic " + base64.StdEncoding.EncodeToString([]byte(login))
```

Comment:
Not a big deal, but `roauth` should probably just be `auth` here as well since it's not read-only anymore.